src/test/java/com/list/todo/services/NotificationServiceTest.java

```diff
@@ -0,0 +1,255 @@
+import com.list.todo.entity.User;
+import com.list.todo.entity.UserSettings;
+import com.list.todo.util.ObjectsProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.list.todo.util.ObjectsProvider.*;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class NotificationServiceTest {
+
+    @Mock
+    private EmailService emailServiceMock;
+
+    @Mock
+    private UserSettingsService userSettingsServiceMock;
+
+    @Mock
+    private SimpMessagingTemplate webSocketMock;
+
+    @Mock
+    private FollowerService followerServiceMock;
+
+    @InjectMocks
+    private NotificationService notificationServiceMock;
+
+    @Test
+    public void notifyAboutSharingTodoList_IfEnableEmailNotification_SuccessfulNotify() {
+        //arrange
+        UserSettings userSettings = UserSettings.builder()
+                .isEnableEmailNotification(true)
+                .isEnableWebSocketNotification(false)
+                .build();
+
+        User ownerUser = createUserWithUserSettings(1, userSettings);
+        User targetUser = createUserWithUserSettings(2, userSettings);
+        TodoList sharedTodoList = createTodoList(ownerUser.getUsername());
+
+        String channel = "/" + targetUser.getName();
+
+        when(userSettingsServiceMock.getUserSettingsByUserId(targetUser.getId())).thenReturn(Optional.of(targetUser.getUserSettings()));
+
+        //act
+        notificationServiceMock.notifyAboutSharingTodoList(ownerUser, targetUser, sharedTodoList);
+
+        //assert
+        verify(emailServiceMock, times(1)).sendEmail(eq(targetUser.getEmail()), anyString(), anyString());
+        verify(webSocketMock, never()).convertAndSend(eq(channel), anyString());
+    }
+
+    @Test
+    public void notifyAboutSharingTodoList_IfEnableWebSocketNotification_SuccessfulNotify() {
+        //arrange
+        UserSettings userSettings = UserSettings.builder()
+                .isEnableEmailNotification(false)
+                .isEnableWebSocketNotification(true)
+                .build();
+
+        User ownerUser = createUserWithUserSettings(1, userSettings);
+        User targetUser = createUserWithUserSettings(2, userSettings);
+        TodoList sharedTodoList = createTodoList(ownerUser.getUsername());
+
+        String channel = "/" + targetUser.getName();
+
+        when(userSettingsServiceMock.getUserSettingsByUserId(targetUser.getId())).thenReturn(Optional.of(targetUser.getUserSettings()));
+
+        //act
+        notificationServiceMock.notifyAboutSharingTodoList(ownerUser, targetUser, sharedTodoList);
+
+        //assert
+        verify(emailServiceMock, never()).sendEmail(eq(targetUser.getEmail()), anyString(), anyString());
```

Comment:
no need for `eq(targetUser.getEmail()`, actually you have to check that it was not invoked at all, not just for that particular email